c++/src/ColumnWriter.cc

```diff
@@ -1038,6 +1038,8 @@ namespace orc {
 
     if (!useDictionary) {
       directLengthEncoder->add(length, numValues, notNull);
+    } else if (dictionary.capacity() < numValues) {
```

Comment:
This looks weird. Is it better to grow its size only when required? I think the map internally can optimize this by reserving a larger number.